pkg/proxy/util/utils.go

```diff
@@ -113,6 +114,51 @@ func GetLocalAddrs() ([]net.IP, error) {
 	return localAddrs, nil
 }
 
+// GetInterfaceWithDefaultRouteIPv4Addrs returns a list of all network addresses
+// on the interface with default route
+func GetInterfaceWithDefaultRouteAddrs(family int) *net.IPNet {
```

Comment:
oh, sorry, I see that Tim suggested "the IP with default route" in one of the comments I linked to, but IMO that doesn't really make sense; kube-proxy already has a concept of "the (default) node IP" (which gets passed to `nftables.NewProxier()`, which comes from the `Node` object and ought to correspond to the IP that the loadbalancers are using to reach the node as well. So we should just use that.

(FTR, I think there are also already functions that do this in `k8s.io/apimachinery/pkg/util/net`)

Reply:
`nodeIP` is a single ip which depends on `s.PrimaryIPFamily`.  What should we do if it is a dual-stack cluster?
https://github.com/kubernetes/kubernetes/blob/12fc215656f670dc211f5817797968d74977877e/pkg/proxy/nftables/proxier.go#L198

Reply:
Each `Proxier` is single-stack and gets passed the node IP for its associated family. So one proxier only gets the IPv4 node IP and only worries about IPv4 nodeport addresses and the other only gets the IPv6 node IP and only worries about IPv6 nodeport addresses.

Reply:
Gocha.

Reply:
Addressed your comments. Just ignore the unit test related code. I'll complete it after we confirmed we are on the right way.